### PR TITLE
New question flow

### PR DESCRIPTION
**Pull request by attekei**
This supersedes the old PR #4642 which pointed to an old branch.

To-do before code review & merge of the first cut:
- [x] sometimes sorting by creator doesn’t work if you have previously visited QB
- [x] sometimes the list is completely empty (trying to find a way to consistently reproduce this)
- [x] add empty states if there are no metrics/segments or search produces an empty result
- [x] implement hover states for all clickable items
- [x] use the segment as a filter as opposed to just applying the filter contained in the segment
- [x] add smoke-y tests for both metrics and segment flow
- [x] update the page title accordingly

### CODE REVIEW

**Comment by mazameli** [comment]
Okay, I combed through the current state of the branch and took notes. I also noted a few simple items that I'll try to fix myself.

- I'm noticing the grid of cards goes to a single row when the screen is wide vs. when the screen is short. I think the behavior should be that we display a 2x2 grid provided the screen is _tall_ enough (i.e. via a min-height media query).
- There's an unnecessary row at the bottom of each list (because of leaving a space for pagination?) — **known issue**
- I think we should add an X icon button at the far-right of the filter box to clear out the filter search term
- After clicking on a different "view by" option, it would be nice if the focus was set back in the search box
- The Table icon is the wrong one. I can fix this.
- The light blue/gray page background color doesn't extend to the bottom of the page. I filed this is #5792. It's an existing problem on our Stats instance with the Dashboards section, too.
- need to add `pointer: cursor` on everything clickable
- `View by` options need a hover state. We can just make it the same green color as the "active" selection.
- The gray we're using for the row hover state is a different gray from the background of each list's heading. We should just use the heading's gray color in both places. I think I can do this for you.
- when I try to click an item in a list when it's broken out by anything other than Name, I always get:
```
EntitySearch.jsx:267 Uncaught TypeError: chooseEntity is not a function
    at SearchResultListItem._this4.onClick (EntitySearch.jsx:267)
    at Object.ReactErrorUtils.invokeGuardedCallback (ReactErrorUtils.js?47d4:73)
    at executeDispatch (EventPluginUtils.js?65c4:89)
    at Object.executeDispatchesInOrder (EventPluginUtils.js?65c4:112)
    at executeDispatchesAndRelease (EventPluginHub.js?fed9:47)
    at executeDispatchesAndReleaseTopLevel (EventPluginHub.js?fed9:58)
    at Array.forEach (<anonymous>)
    at forEachAccumulated (forEachAccumulated.js?6bd0:28)
    at Object.processEventQueue (EventPluginHub.js?fed9:261)
    at runEventQueueInBatch (ReactEventEmitterMixin.js?0c5b:21)
```
- Need to tweak the padding for the start page cards because the explanatory text feels cramped in the single-row view, and wraps oddly in the 2x2 view. I can take care of this to-do.
- Need to play with the font size for the metric/segment titles in lists. I'll do this.
- I think we should change the title of the "new question" card to "Custom" as long as at least 1 segment or 1 metric exists. 
- We still need to dial in the card layout so that this state isn't possible:

screen shot 2017-08-22 at 12 30 03 pm

**Possible enhancements**
- I think I want to revisit the idea of being able to use the arrow keys to select an item. We backed away from it because there were some weird edge cases, but I think it would feel really nice.
- Do we want to think about adding another card for "new question from saved question?" With a better label than that of course.

**Comment by mazameli** [comment]
**Polish committee to-do list**

**Atte**
- [ ] fix bug where cards disappear under the nav bar
- [ ] when clearing out filter input with X button, input should regain focus

**Maz**
- [ ] dial in the `custom` option illustration 
- [ ] dial in the SQL illustration (borders are too thick, looks different)
- [ ] dial in the responsive sizes for cards
- [ ] dial in height of pagination row
- [ ] dial back contrast/visual prominence of pagination
- [ ] dial in font size of metric titles vs. the view by options
- [ ] cut out descriptions from lists
- [ ] view-by filters feel overly spaced out
- [ ] make the list grouping icon gray color bluer

**Possible enhancements:**
- after clicking on a metric, going back could remember the filtering you did
- command-click to open metric/segment item in new tab

File: frontend/src/metabase/new_query/containers/MetricSearch.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        .value()
+
+                    if (sortedActiveMetrics.length > 0) {
+                        return (
+                            <EntitySearch
+                                title="Which metric?"
+                                // TODO Atte Keinänen 8/22/17: If you call `/api/table/:id/table_metadata` it returns
+                                // all metrics (also retired ones) and is missing `is_active` prop. Currently this
+                                // filters them out but we should definitely update the endpoints in the upcoming metadata API refactoring.
+                                entities={sortedActiveMetrics}
```
Comment (salsakran) [inline frontend/src/metabase/new_query/containers/MetricSearch.jsx]: this should be a fairly quick backend endpoint to bang out, right @camsaul ?

though in general, I'd like to be a little saner about what gets pulled down onto the frontend redux store.

**Review by salsakran — COMMENTED** [review]
@attekei what needs to be loaded before the illustrations appear?

ideally, there should be no network calls between clicking the new question button and that first screen.

**Comment by attekei** [comment]
In the first screen there are two conditionals:
- If the user is admin, then show all buttons (those could definitely be shown immediately)
- If the user isn't an admin, then show metrics button if the user has access to >0 metrics and segments button if the user has access to >0 segments. That requires two API calls (although the stupidity here is that we only need info about the count of available metrics/segments but actually pull the complete metadata about them)

**Comment by salsakran** [comment]
@camsaul can you help out here? It's not really acceptable to have to pull down _all_ the metadata we have (including tables, metrics, segments and saved questions), to display 4 illustrations.

**Comment by salsakran** [comment]
i.e. we can merge this mini-cycle but before 0.26 goes out, this should become a super snappy click.

**Comment by attekei** [comment]
I modified the start screen so that it shows all buttons immediately if the user is admin, but as @salsakran said we should have a more lightweight way to fetch required information to show the relevant buttons for non-admin users.

**Review by kdoh — APPROVED** [review]

**Pull request merged**